app/redux/services/unlockWallet.ts

```diff
@@ -19,18 +20,23 @@ export const unlockWallet = async (password: string, startInOfflineMode = false)
 
   const { walletStatus } = await fullServiceApi.getWalletStatus();
 
-  const firstAccountId = walletStatus.accountIds[0];
+  const accounts: Accounts = {
+    accountIds: walletStatus.accountIds ?? [],
+    accountMap: walletStatus.accountMap ?? {},
+  };
 
-  let { selectedAccount, addingAccount } = initialReduxStoreState;
+  const firstAccountId = (walletStatus.accountIds ?? [])[0];
 
+  let { selectedAccount, addingAccount } = initialReduxStoreState;
   // if an account already exists, use the default to the first account available. Otherwise, use the initial state
   if (firstAccountId) {
     const { balance: balanceStatus } = await fullServiceApi.getBalanceForAccount({
       accountId: firstAccountId,
     });
 
+    const account = (walletStatus.accountMap ?? {})[firstAccountId];
```

Comment:
The default value of `{}` doesn't have the property `[firstAccountId]` and will cause unhandled errors if that occurs. Maybe something like this:
```suggestion
    if (walletStatus.accountMap && firstAccountId in walletStatus.accountMap) {
        const account = walletStatus.accountMap[firstAccountId]
    } else {
        const account = {}
    }
```

Reply:
yeah, I will fix this. going to add `const firstAccount = firstAccountId && (walletStatus.accountMap ?? {})[firstAccountId];` and then check for it's presence in the same conditional where I check for firstaccountid